sdk/samples/core/mosquitto_rpc_client_sample.c

```diff
@@ -247,55 +289,27 @@ int main(int argc, char* argv[])
   az_result rc;
 
   // infinite execution loop
-  for (int i = 45; !sample_finished && i > 0; i++)
+  for (int i = 0; !sample_finished; i++)
   {
-    // remove any expired commands
-    pending_command* expired_command = get_first_expired_command(pending_commands);
-    while (expired_command != NULL)
-    {
-      printf(LOG_APP_ERROR "command ");
-      print_correlation_id(expired_command->correlation_id);
-      printf(" expired\n");
-      az_result ret = remove_command(&pending_commands, expired_command->correlation_id);
-      if (ret != AZ_OK)
-      {
-        printf(LOG_APP_ERROR "Expired command not a pending command\n");
-      }
-      expired_command = get_first_expired_command(pending_commands);
-    }
-#ifdef _WIN32
-    Sleep((DWORD)1000);
-#else
-    sleep(1);
-#endif
     printf(LOG_APP "Waiting...\r");
     fflush(stdout);
+
+    // remove any expired commands from the client
+    remove_expired_commands();
```

Comment:
This expiration is computed on the client-side. We could add a comment that, even for expired commands, we will be calling the callback with results that are received from the other end.